discovery-provider/alembic/versions/92571f94989a_add_trending_views.py

```diff
@@ -0,0 +1,290 @@
+        -- -- join on subquery for saves by year
+        LEFT OUTER JOIN (
+            SELECT
+                r.save_item_id as track_id,
+                count(r.save_item_id) as repost_count
+            FROM
+                saves r
+            WHERE
+                r.is_current is True AND
+                r.save_type = 'track' AND
+                r.is_delete is False AND 
+                r.created_at > (now() - interval '1 year')
+            GROUP BY r.save_item_id
+        ) save_year ON save_year.track_id = t.track_id
+        -- -- join on subquery for saves by month
+        LEFT OUTER JOIN (
+            SELECT
+                r.save_item_id as track_id,
+                count(r.save_item_id) as repost_count
+            FROM
+                saves r
+            WHERE
+                r.is_current is True AND
+                r.save_type = 'track' AND
+                r.is_delete is False AND 
+                r.created_at > (now() - interval '1 month')
+            GROUP BY r.save_item_id
+        ) save_month ON save_month.track_id = t.track_id
+        -- -- join on subquery for saves by week
+        LEFT OUTER JOIN (
+            SELECT
+                r.save_item_id as track_id,
+                count(r.save_item_id) as repost_count
+            FROM
+                saves r
+            WHERE
+                r.is_current is True AND
+                r.save_type = 'track' AND
+                r.is_delete is False AND 
+                r.created_at > (now() - interval '1 week')
+            GROUP BY r.save_item_id
+        ) save_week ON save_week.track_id = t.track_id
+        LEFT OUTER JOIN (
+            SELECT
+                save_and_reposts.item_id as track_id,
+                sum(au.follower_count) as karma
+            FROM
+                (
+                    select 
+                        r_and_s.user_id, 
+                        r_and_s.item_id
+                    from 
+                        (select 
+                            user_id, 
+                            repost_item_id as item_id
+                        from 
+                            reposts
+                        where
+                            is_delete is false AND
+                            is_current is true AND
+                            repost_type = 'track'
+                        union all
+                        select 
+                            user_id, 
+                            save_item_id as item_id
+                        from 
+                            saves
+                        where
+                            is_delete is false AND
+                            is_current is true AND
+                            save_type = 'track'
+                        ) r_and_s
+                    join 
+                        users
+                    on r_and_s.user_id = users.user_id
+                    where
+                        users.cover_photo is not null AND
```

Comment:
with this change here, I'm not sure how we can experiment/ever change trending again. would we have to create a new migration with a whole new table?

Reply:
Karma seems like an outlier here - the rest of the variables seem straightforward. Maybe a separate matview for user karma?

Reply:
point mostly being that we may want to add other things like karma in the future, and it will be quite difficult for us to experiment here. i think this is reasonable for now though given that the other stuff is changeable

Reply:
For experiments and adding more factors to the trending score, we would update this mat view w/ the data we need. This only takes ~11 seconds to update and applies to ALL 400k tracks - which is already better than the current system which only does it for 500 tracks. I don't this it's too hard to write the logic in SQL compared w/ python and it's pretty powerful in terms of what it can do. 
I was debating over moving karma over to another view, but I don't think it'll make too much of a difference.